src/us/kbase/narrativemethodstore/db/mongo/MongoUtils.java

```diff
@@ -64,6 +71,20 @@ public static DBObject toDBObject(final Object obj) {
         return MAPPER.convertValue(obj, new TypeReference<Map<String, Object>>() {});
     }
     
+    
+    /** Map a MongoDB {@link DBObject} to a class.
+     * This method expects that all maps and lists in the objects are implemented as
+     * {@link BSONObject}s or derived classes, not standard maps, lists, or other classes.
+     * The object must be deserializable by an {@link ObjectMapper} configured so private
+     * fields are visible.
+     * @param dbo the MongoDB object to transform.
+     * @param clazz the class to which the object will be transformed.
+     * @return the transformed object.
+     */
+    public static <T> T toObject(final DBObject dbo, final Class<T> clazz) {
+        return dbo == null ? null : MAPPER.convertValue(toMap(dbo), clazz);
```

Comment:
clazz is great. I'm assuming java folk don't use underscore prefixes to avoid key words?

Reply:
Generally no, although you do see it every now and then. In this case `clazz` is the nom de guerre. Also there are far fewer keywords in java than in python, so avoiding keywords is a pretty rare problem. `public` and `private` are the ones I hit the most.